WordPress/Classes/ViewRelated/Menus/Views/MenusSelectionView.h

```diff
@@ -0,0 +1,74 @@
+#import <UIKit/UIKit.h>
+#import "MenusSelectionItem.h"
+
+typedef NS_ENUM(NSUInteger, MenusSelectionViewType) {
+    MenusSelectionViewTypeMenus = 1,
+    MenusSelectionViewTypeLocations
+};
+
+@protocol MenusSelectionViewDelegate;
+
+@interface MenusSelectionView : UIView
+
+@property (nonatomic, weak) id <MenusSelectionViewDelegate> delegate;
+
+/**
+ The type of selection the selectionView should be configured for.
+ */
+@property (nonatomic, assign) MenusSelectionViewType selectionType;
+
+/**
+ Toggle the visual expansion of the selection items in the UI.
+ */
+@property (nonatomic, assign) BOOL selectionItemsExpanded;
+
+/**
+ The currently selected item in the UI.
+ */
+@property (nonatomic, strong) MenusSelectionItem *selectedItem;
+
+/**
+ Add a selection item to the list of available items for display.
+ */
+- (void)addSelectionViewItem:(MenusSelectionItem *)selectionItem;
+
+/**
+ Remove a selection item from the list of available items for display.
+ */
+- (void)removeSelectionItem:(MenusSelectionItem *)selectionItem;
+
+/**
+ Remove all selection items from the list of available items for display.
+ */
+- (void)removeAllSelectionItems;
+
+/**
+ Create a new selection item with an associated object the item represents.
+ */
+- (MenusSelectionItem *)itemWithItemObjectEqualTo:(id)itemObject;
+
+/**
+ Toggle the visual expansion of the selection items in the UI, with animation.
+ */
+- (void)setSelectionItemsExpanded:(BOOL)selectionItemsExpanded animated:(BOOL)animated;
+
+@end
+
+@protocol MenusSelectionViewDelegate <NSObject>
+
+/**
+ The user tapped or pressed to toggle the selection expansion of the UI.
+ */
+- (void)userInteractionDetectedForTogglingSelectionView:(MenusSelectionView *)selectionView expand:(BOOL)expand;
```

Comment:
Just a suggestion.

I'd look for a way to make "user interaction detected for toggling selection view" (the method name) be a bit less indirect.

Maybe use something like `userTappedOnSelectionView:expand:`, or something similar?